api/cluster/resource/utils.go

```diff
@@ -43,3 +52,19 @@ func toKServeDeploymentMode(deploymentMode deployment.Mode) (string, error) {
 		return "", fmt.Errorf("unsupported deployment mode: %s", deploymentMode)
 	}
 }
+
+func computeResource(resourceQuantity resource.Quantity, fraction float64) resource.Quantity {
+	scaledValue := resourceQuantity.Value()
+
+	scaledMilliValue := int64(math.MaxInt64 - 1)
+	if scaledValue < (math.MaxInt64 / 1000) {
+		scaledMilliValue = resourceQuantity.MilliValue()
+	}
+	percentageValue := float64(scaledMilliValue) * fraction
+	newValue := int64(math.MaxInt64)
+	if percentageValue < math.MaxInt64 {
+		newValue = int64(percentageValue)
+	}
+	newquantity := resource.NewMilliQuantity(newValue, resource.BinarySI)
+	return *newquantity
+}
```

Comment:
Same for this which is linked to https://github.com/knative/serving/blob/release-0.14/pkg/reconciler/revision/resources/queue.go#L115